.github/workflows/rust.yml

```diff
@@ -106,27 +98,41 @@ jobs:
           override: true
           profile: minimal
           components: rustfmt, clippy
-      - uses: actions-rs/install@v0.1
+      - name: Install cargo-workspaces
+        uses: actions-rs/install@v0.1
+        with:
+          crate: cargo-workspaces
+      - name: Install cargo-fuzz
+        uses: actions-rs/install@v0.1
         with:
           crate: cargo-fuzz
           version: latest
       - uses: Swatinem/rust-cache@v2
         with:
           key: misc
+
       - name: Format (rustfmt)
         run: cargo fmt --all --check
+
       - name: Lint (All features)
-        run: cargo clippy --all-features --all-targets
+        run: cargo workspaces exec cargo clippy --all-features --all-targets -- -D warnings
       - name: Lint (No features)
-        run: cargo clippy -p boa_engine --no-default-features --all-targets
+        run: cargo workspaces exec cargo clippy --no-default-features --all-targets -- -D warnings
+
       - name: Generate documentation
         run: cargo doc -v --document-private-items --all-features
-      - name: Build
-        run: cargo build --all-targets --quiet --profile ci
+
+      - name: Build (All features)
```

Comment:
Ooooh, definitely like this better. While making this large of an update to the CI. Should `Misc` be broken apart into more firmly labelled tests? For example, instead of `Misc` there could be tests labelled `Formatting and Lints` and `Crate Builds`.

Reply:
Definitely! We can't separate build + build fuzz + test examples since those are done that way to save building time, but we can definitely split the rest into smaller jobs.

Reply:
Oh, apparently the fuzzer CAN be separated, since it triggers a rebuild anyways because it uses the `fuzz` feature only.